### PR TITLE
chore: purify the junit reporter

### PR DESCRIPTION
**Pull request by pavelfeldman**
https://github.com/microsoft/playwright/pull/11374 introduced a number of configuration options for JUnit reporter that was targeting a better alignment with the [Xray JUnit dialect](https://docs.getxray.app/display/XRAYCLOUD/Taking+advantage+of+JUnit+XML+reports).

We are seeing more Xray-specific requests to JUnit:
- https://github.com/microsoft/playwright/issues/22396
- https://github.com/microsoft/playwright/issues/22596

As well as the requests to counter the Xray changes:
- https://github.com/microsoft/playwright/issues/18379

When looking at https://github.com/microsoft/playwright/issues/22396, it is clear that we are setting up Xray user to failure: instead of generating a full-fledged Xray report, we are going through an intermediate JUnit format where we are hacking information in a lossy manner. What's worse is that we are polluting JUnit with Xray-specific options that pretend to be generic ones, without those actually being requested by the JUnit customers.

I'm suggesting that we prune the Xray-specificity from JUnit and allow Xray team or third parties to implement a full Xray reporter for Playwright.

### CODE REVIEW

**Comment by pavelfeldman** [comment]
// cc @bitcoder

**Comment by bitcoder** [comment]
Hi, sorry to hear that but I totally understand and respect your decision.
We'll need a bit of time to move this to a separate reporter. To make it easier right now, maybe the best is to replicate the current behaviour along with the respective code with minor tweaks if needed.
I'll let you know about the progress.

**Comment by pavelfeldman** [comment]
> Hi, sorry to hear that but I totally understand and respect your decision. We'll need a bit of time to move this to a separate reporter. To make it easier right now, maybe the best is to replicate the current behaviour along with the respective code with minor tweaks if needed. I'll let you know about the progress.

I won't be landing this for now and will ping you in a couple of weeks!

**Comment by bitcoder** [comment]
@pavelfeldman can I kindly ask you to quickly review the following project?
I created it using mostly the existing code and did some subtle changes to make it compile and have the tests passing.
I kept the license notice on the files copied from this project.
It's not something perfect but I think it works but I appreciate if you can give it a look

https://github.com/Xray-App/playwright-junit-reporter

**Comment by pavelfeldman** [comment]
> Would it be possible to maintain the API that supports adding these properties in the JUnit reporter? I.e.:

@d-rede, that makes sense, so we would make it work like `embedAnnotationsAsProperties` is true by default. Are you also interested in attachments or annotations only?

File: packages/playwright-test/src/reporters/junit.ts [hunk 1]
```diff
@@ -31,17 +31,10 @@ class JUnitReporter implements Reporter {
   private totalSkipped = 0;
   private outputFile: string | undefined;
   private stripANSIControlSequences = false;
-  private embedAnnotationsAsProperties = false;
-  private textContentAnnotations: string[] | undefined;
-  private embedAttachmentsAsProperty: string | undefined;
 
-
-  constructor(options: { outputFile?: string, stripANSIControlSequences?: boolean, embedAnnotationsAsProperties?: boolean, textContentAnnotations?: string[], embedAttachmentsAsProperty?: string } = {}) {
+  constructor(options: { outputFile?: string, stripANSIControlSequences?: boolean, embedAnnotationsAsProperties?: boolean, textContentAnnotations?: string[] } = {}) {
```
Comment (dgozman) [inline packages/playwright-test/src/reporters/junit.ts:35]:
```suggestion
  constructor(options: { outputFile?: string, stripANSIControlSequences?: boolean } = {}) {
```

**Review by dgozman — APPROVED** [review]

**Comment by bitcoder** [comment]
> @pavelfeldman can I kindly ask you to quickly review the following project? I created it using mostly the existing code and did some subtle changes to make it compile and have the tests passing. I kept the license notice on the files copied from this project. It's not something perfect but I think it works but I appreciate if you can give it a look
> 
> https://github.com/Xray-App/playwright-junit-reporter

Did you have opportunity to give it a look? Would appreciate your feedback :)

**Comment by pavelfeldman** [comment]
I looked briefly and it seemed fine!

**Pull request merged**